### PR TITLE
feat: use tz_offset when doing deployment dau query

### LINKED ISSUE #7275: bug: daily active users modal in Deployment tab is incorrect
customer daily active user count showed 16 on Sunday, which was incorrect, since no users had logged in. the figure was likely off by a day, since it showed 0 for the previous day, Saturday.

### PR DESCRIPTION
**Pull request by Emyrk**
Fixes https://github.com/coder/coder/issues/7275

### CODE REVIEW

File: coderd/database/dbfake/databasefake.go [hunk 1]
```diff
@@ -453,7 +453,7 @@ func (q *fakeQuerier) GetTemplateDAUs(_ context.Context, arg database.GetTemplat
 			continue
 		}
 
-		date := as.CreatedAt.UTC().Add(time.Duration(arg.TzOffset) * time.Hour).Truncate(time.Hour * 24)
+		date := as.CreatedAt.UTC().Add(time.Duration(arg.TzOffset) * time.Hour * -1).Truncate(time.Hour * 24)
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/database/dbfake/databasefake.go:456]: You are half right. 

I forgot to change it in the Dau dbfake function, which does break the unit test.

The template test does not break because I reverted the Template DAU to only use `UTC` again for performance reasons. So this is always `0` as the offset. 


----

tl;dr fixing it in the spot that is used **does break the test**

**Review by johnstcn — APPROVED** [review]
